File: Exercism/ruby/difference-of-squares/difference_of_squares.rb
# frozen_string_literal: true

# Squares ...
class Squares
  attr_accessor :square_of_sum
  attr_accessor :sum_of_squares
  attr_accessor :difference

  def initialize(num)
    @square_of_sum = (1..num).sum**2
    @sum_of_squares = (1..num).map { |x| x**2 }.sum
    @difference = @square_of_sum - @sum_of_squares
  end
end

File: Exercism/fortran/difference-of-squares/difference_of_squares.f90
module difference_of_squares
   implicit none
contains

   integer function square_of_sum(n)
      integer :: n, i
      integer, dimension(n) :: arr
      arr = (/(i, i=1, n)/)
      square_of_sum = sum(arr)**2
   end function square_of_sum

   integer function sum_of_squares(n)
      integer :: n, i
      integer, dimension(n) :: arr
      arr = (/(i**2, i=1, n)/)
      sum_of_squares = sum(arr)
   end function sum_of_squares

   integer function difference(n)
      integer :: n
      difference = square_of_sum(n) - sum_of_squares(n)
   end function difference

end module difference_of_squares
